zerver/views/drafts.py

```diff
@@ -85,11 +87,22 @@ def further_validated_draft_dict(draft_dict: Dict[str, Any],
         "last_edit_time": last_edit_time,
     }
 
+def draft_endpoint(view_func: ViewFuncT) -> ViewFuncT:
+    @wraps(view_func)
+    def draft_view_func(request: HttpRequest, user_profile: UserProfile,
+                        *args: object, **kwargs: object) -> HttpResponse:
+        if not user_profile.drafts_syncing_enabled:
+            return json_error(_("User has not enabled drafts syncing."))
+        return view_func(request, user_profile, *args, **kwargs)
+    return cast(ViewFuncT, draft_view_func)  # https://github.com/python/mypy/issues/1927
```

Comment:
Hmm, what's your thinking on having this be a server-side access check, vs. a client-side decision on whether to try to send drafts to the server?  I'd generally expect this sort of check to be client-side, but I could could see an argument to go the other way.  (Do we want to have the `do_...` function in `actions.py` remove all existing drafts when one disables the synchronization setting?  Probably)

In any case, we probably want the `human_users_only` decorator on these views as well (to disable bot access).

Reply:
@timabbott  good question! There will definitely be client side checks for this but I've added a server side check as well as a sort of defensive measure.

[Feedback](https://chat.zulip.org/#narrow/stream/2-general/topic/drafts/near/931751) from @showell made me realize that if someone doesn't want to sync their drafts, they can explicitly tell the server they don't want drafts to be synced. This way even if a client is kinda accidentally buggy then drafts won't be synced (though this isn't the case if it straight up ignores what the user wants since it can always turn this flag back on).

Reply:
> Do we want to have the do_... function in actions.py remove all existing drafts when one disables the synchronization setting? Probably

Yup, and that's exactly what we're doing. Ref: `do_disable_drafts_syncing` from `zerver.lib.drafts`

Reply:
> In any case, we probably want the human_users_only decorator on these views as well (to disable bot access).

I've actually thought about this before and I think it might be better to allow bots to have their own drafts. This will allow one to get more hackish with what they can do with a bot (which is actually a good thing).

Think about it this way: drafts can be used like templates since the message sending and drafts deletion logic are not really coupled (you could send a message but you don't have to delete the draft - it all depends on the client). So theoretically, one could use drafts as a way of creating message templates for bots which will be stored on the server instead of locally. Following this line of thought, I felt that it would be better to let bots access this endpoint too.

Just to be clear: I don't have a **super** concrete use case for allowing bots to use drafts but rather I have a general feeling that it would be better.